### PR TITLE
winrar: Add 'helper' package

### PR DESCRIPTION
**Pull request by issaclin32**
Will be used in #7388. (as a workaround for the extraction failure mentioned in #7338)

This adds an **WinRAR** "helper" package, which is identical to `winrar` package, but without bin/shortcut.

This is an alternative to `"depends:extras/winrar"`, as it can avoid some confusing situations for end-users.
Most people have their WinRAR installed through other means. If we silently install another WinRAR without asking the user, people may complain about it.
(Why there are 2 WinRARs? Why my WinRAR suddenly turn into English rather than my localized version?)

With this helper package, we can extract the files using *WinRAR* without affecting the users, and always keep this "helper" WinRAR updated.

### CODE REVIEW

**Comment by phanirithvij** [comment]
I opened a duplicate one #7389, please feel free to close one of these two.

**Comment by issaclin32** [comment]
Waiting for opinions from other maintainers.

**Comment by Valinor** [comment]
> Will be used in #7388. (as a workaround for the extraction failure mentioned in #7338)
> 
> This adds an **WinRAR** "helper" package, which is identical to `winrar` package, but without bin/shortcut.
> 
> This is an alternative to `"depends:extras/winrar"`, as it can avoid some confusing situations for end-users. Most people have their WinRAR installed through other means. If we silently install another WinRAR without asking the user, people may complain about it. (Why there are 2 WinRARs? Why my WinRAR suddenly turn into English rather than my localized version?)
> 
> With this helper package, we can extract the files using _WinRAR_ without affecting the users, and always keep this "helper" WinRAR updated.

This is a great improvement, this silent use of winrar. What about the licence (shareware). Can this be a problem?

**Comment by issaclin32** [comment]
> What about the licence (shareware). Can this be a problem?

The license problem is somewhat unsettled.
WinRAR **explicitly** says it is a Shareware. However, it does not put any restrictions on free users, except for a message box.

I think we can leave this for discussion in another issue/PR.

**Comment by rashil2000** [comment]
The idea seems fine to me, but won't https://github.com/ScoopInstaller/Scoop/pull/3502 be a better and more robust solution for this? Then we don't need to depend on anything.

**Comment by issaclin32** [comment]
> The idea seems fine to me, but won't [ScoopInstaller/Scoop#3502](https://github.com/ScoopInstaller/Scoop/pull/3502) be a better and more robust solution for this? Then we don't need to depend on anything.

Yeah. It should be a better solution. But we will still need this `winrar-helper` package, since some NSIS installer now needs WinRAR for extraction.

I tend to add `"depends": "winrar-helper"` and use *pre_install* script as a quick fix. We can delete it from the manifest after built-in unpacker is settled and merged into *master*.

**Comment by niheaven** [comment]
`Invoke-ExternalCommand winrar.exe @('x', '-ibck', '"F:\Downloads\Scratch 3.27.0 Setup.exe"', 'F:\Downloads\scratch\')`

Emm, then what's wrong with Scratch's installer?

**Comment by niheaven** [comment]
I'll add it as fallback in https://github.com/ScoopInstaller/Scoop/pull/3502

**Comment by issaclin32** [comment]
Base on the comments, I assume this package is ready to go.

**Comment by rashil2000** [comment]
OK from me 👍

**Review by rashil2000 — APPROVED** [review]

**Pull request merged**
